### DISCUSSION TITLE
When the python allTests.py-d command is executed in the cpp directory in the ARM architecture, the following error message

### DISCUSSION THREAD

**Discussion by lillianokok** [zeroc-ice/ice#3306] [Q&A] [answered]
### Version

3.7.10

### Describe the bug

When the python allTests.py-d command is executed in the cpp directory in the ARM architecture, the following error message is displayed:
Test Certificate Verification...src/IceSSL/OpenSSLTransceiverI.cpp:444:::Ice::SecurityException:
Security Exceptions:
IceSSL: Certificate validation failed:
The certificate has expired.
Failure!
test/IceSSL/configuration/AllTests.cpp:717: Assertion "false" failed
-!12/25/24 10:55:29.216/home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration/build/aarch64/shared/ Server: Warning: Connection Exception:
src/Ice/StreamSocket.cpp:213: ::Ice::ConnectionLostException:
Connection lost: recv() returns zero
0 IceUtil:: Exception:: /home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration/build/aarch64/shared/../../../../../../../../..//...//lib64/libIce.so.37
1 Ice:::LocalException::/home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration/build/aarch64/shared/../../../../../../../../../..//...//lib64/libIce.so.37
2 Ice::SyscallException::SyscallException(char const*, int, int) at /home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration/build/aarch64/shared/../../../../../../../../../...//lib64/libIce.so.37
3 Ice:::SocketException::/home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/ SocketException(char const*, int, int)/configuration/build/aarch64/shared/../../../../../../../../../..../lib64/libIce.so.37 in test/IceSSL
4 Ice:::ConnectionLostException::/home/stage/root/spack-stage-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration/build/aarch64/shared/../../../../../../../../../...//lib64/libIce.so.37
5 IceInternal::StreamSocket::/home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/c read(char*, unsigned long)/build/aarch64/shared/../../../../../../../../../...//lib64/libIce.so.37 in pp/test/IceSSL/configuration
6 IceInternal::StreamSocket::read(IceInternal::Buffer&) in /home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration /build/aarch64/shared/../../../../../../../../../...//lib64/libIce.so.37
7/home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration/build/aarch64/shared/../../../.../lib 64/libIce.so.37(+0x178334)[0xffffb60d5334]
8/home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration/build/aarch64/shared/../../../.../lib 64/libIce.so.37(+0x18255c)[0xffffb60df55c]
9/home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration/build/aarch64/shared/../../../.../lib 64/libIce.so.37(+0x292a38)[0xffffb61efa38]
10/home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration/build/aarch64/shared/../../../.../lib 64/libIce.so.37(+0x2933d4)[0xffffb61f03d4]
11/home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration/build/aarch64/shared/../../../.../lib 64/libIce.so.37(+0x28cce0)[0xffffb61e9ce0]
12 /usr/lib64/libc.so.6(+0x822a8) [0xffffb5b682a8]
13 /usr/lib64/libc.so.6(+0xe9a1c) [0xffffb5bcfa1c]

Local Address = 127.0. 0.1:12010
Remote Address = 127.0. 0.1:54248
-!12/25/24 10:55:29.217/home/stage/root/spack-stage-ice-3.7.10-oshfhozsdq4jg53zdo5cgovqxiklrdoy/spack-src/cpp/test/IceSSL/configuration/build/aarch64/shared/ Server: Warning: Connection Exception:
src/Ice/StreamSocket.cpp:213: ::Ice::ConnectionLostException:
Connection lost: recv() returns zero

### Expected behavior

no errors

**Comment by pepone** [comment, marked as answer]
Hi,

The 3.7.10 SSL test certificates expired recently. We have updated the certificates in the [3.7](https://github.com/zeroc-ice/ice/tree/3.7) branch.

See:

52da60d56bcddf1dc8fe7970d8116f030dc01434
234b98676be05280b8843871f3af9ee5a666fcf6

Cheers,
Jose
